src/main/scala/view/GameScreen.scala

```diff
@@ -134,7 +120,9 @@ class GameScreen() extends ScreenAdapter with EntityRenderer :
 
   def memoizedTexture: Entity => Texture =
 
-    def texture(entity: Entity): Texture = new Texture(Gdx.files.classpath("assets/" + spriteName(entity)))
+    def texture(entity: Entity): Texture =
+      val spritename = spriteName(entity)

```

Comment:
I don't get why you are saving the sprite name and then you're not using it (by the way the val name should be 'spriteName').
I would revert the changes and leave how it was before.

Reply:
Right, just forgot to delete the variable